docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -310,37 +310,39 @@ summarize the distribution differently or offer more configuration options.
 
 ### Best practices
 
-- Histograms tend to store a lot more data in memory than other metric types, however, the exact memory usage is determined by the collection tool being used.
+- Histograms tend to store a lot more data in memory than other metric types. However, the exact memory usage is determined by the collection tool being used.
   If you're defining a large number (>100) of Histogram metrics, you may need to give users guidance not to enable them all at the same time, or to configure their tools to save
   memory by reducing precision. Some collection tools may have hard limits on the number of concurrent Histograms they will monitor to prevent excessive memory use.
 
 - Callbacks for all observable instruments are invoked in sequence, so any callback that takes a long time can delay or prevent all metrics from being collected. Favor
   quickly reading a cached value, returning no measurements, or throwing an exception over performing any potentially long-running or blocking operation.
 
+- The ObservableCounter and ObservableGauge instrument types don't guarantee thread safety. The callbacks occur on a thread that's not usually synchronized with the code that updates the values. When the callback functions are called by multiple threads, it's your responsibility to guarantee the delegates' thread safety.
```

Comment:
```suggestion
- The ObservableCounter, ObservableUpDownCounter, and ObservableGauge  callbacks occur on a thread that's not usually synchronized with the code that updates the values. It is your responsibility to either synchronize memory access or accept the inconsistent values that may be observed when using unsynchronized access. Common approaches to synchronize access are using a lock or using <xref:System.Threading.Volatile.Write> and <xref:System.Threading.Volatile.Read>. 
```

What do you think about this wording?

Reply:
Looks good, but maybe change last sentence to:

"Common approaches to synchronize access are to use a lock or call <xref:System.Threading.Volatile.Write> and <xref:System.Threading.Volatile.Read>. "

Reply:
@noahfalk @gewarren - How about another tweak: "It's your responsibility to either synchronize memory access or accept the inconsistent values that can result from using unsynchronized access."

Reply:
sounds fine to me!